scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -284,6 +288,35 @@ object Execution {
   }
 
   /**
+    * This is a Thread used as a shutdown hook to clean up temporary files created by some Execution
+    *
+    * If the job is aborted the shutdown hook may not run and the temporary files will not get cleaned up
+    */
+  private[scalding] case class TempFileCleanup(filesToCleanup: Iterable[String], mode: Mode) extends Thread {
+    val LOG = LoggerFactory.getLogger(this.getClass)
+
+    override def run(): Unit = {
+      val fs = mode match {
+        case localMode: CascadingLocal => FileSystem.getLocal(new Configuration)
+        case hdfsMode: HadoopMode =>  FileSystem.get(hdfsMode.jobConf)
+      }
+
+      filesToCleanup.foreach { file: String =>
+        try {
+          val path = new Path(file)
+          if (fs.exists(path)) {
+            // The "true" parameter here indicates that we should recursively delete everything under the given path
+            fs.delete(path, true)
+          }
+        } catch {
+          // If we fail in deleting a temp file, log the error but don't fail the run
+          case e: Throwable => LOG.warn(s"Unable to delete temp file $file", e)
```

Comment:
It is the same, but leaving out the `: Throwable` generates a compiler warning requesting that you be explicit. I see a mix of both in the codebase now, but I'm happy to go with whatever's the preferred style.